### PR TITLE
Observable name refactor

### PR DESCRIPTION
**Pull request by johannesjmeyer**
**Description of the Change:**
Rename the old expectations (e.g. MeanPhoton) to the corresponding observables they represent (e.g. NumberOperator).

**Benefits:**
Naming will be more coherent and logical.

**Possible Drawbacks:**
Renaming the widely used observables is a breaking change -- which is okay since compatibility was already broken by #232 

**Related GitHub Issues:**
#243

### CODE REVIEW

**Comment by johannesjmeyer** [comment]
I also replaced the observable `NumberState` with `NumberStateProjector` to avoid confusion with a number state preparation which is `FockState`, we can discuss about the name though. 

But I think we should decide if we want to talk about Fock states or Number states, mixing those two seems quite irritating. What is your opinion on this?

**Comment by josh146** [comment]
I agree that `NumberStateProjector` makes more sense then just `NumberState`, however I am biased towards shorter names for conciseness.

> But I think we should decide if we want to talk about Fock states or Number states, mixing those two seems quite irritating. What is your opinion on this?

Post #232, we now support a combined observable/operator. Does it make sense to have `qml.FockState` inherit from both `CVOperation` and `CVObservable`? When applied within a circuit, it acts as a Fock state preparation operation, when returned with `qml.expval(qml.FockState(2))`, it acts as a projector?

This could be a bit of an abuse of terminology, but it streamlines the CV naming a bit.

Just a note: since we are currently undergoing a tutorial and documentation refactor, it is best not to modify any file in the `doc/tutorials` and `examples/` folder, as these are being replaced/rewritten, and will lead to large merge conflicts.

**Comment by johannesjmeyer** [comment]
I see your point, but I'd rather choose a longer name than an ambiguous one. 

In my opinion, combining the two would not be a good idea, because they don't represent the same operator -- FockState is principially |0><n| while NumberStateProjector is |n><n|. Furthermore, I think that `qml.expval(qml.FockState, ...)` reads like we would take the expectation value with the system being in a Fock state.

What do you think about just renaming it from NumberStateProjector to FockStateProjector? (or the other way round)

File: .github/CHANGELOG.md [hunk 1]
```diff
@@ -51,6 +51,9 @@
 
 ### Breaking changes
 
+* The following CV observables were renamed to comply with the new Operation/Observable
+  scheme: `Mean photon` to `NumberOperator`, `Homodyne` to `QuadratureOperator` and `NumberState` to `NumberStateProjector`. [#243](https://github.com/XanaduAI/pennylane/pull/243)
```
Comment (co9olguy) [inline .github/CHANGELOG.md]:
```suggestion
  scheme: `MeanPhoton` to `NumberOperator`, `Homodyne` to `QuadratureOperator` and `NumberState` to `NumberStateProjector`. [#243](https://github.com/XanaduAI/pennylane/pull/243)
```

**Comment by co9olguy** [comment]
Sorry about being slow to get to this. There will be a few merge conflicts to resolve (hopefully straightforward)

**Comment by josh146** [comment]
Before merging, a few things that I would like to finalize:

* Let's rename `QuadratureOperator`  to `QuadOperator` - this matches the terminology used in [OpenFermion](https://openfermion.readthedocs.io/en/latest/openfermion.html#openfermion.ops.QuadOperator) and [Strawberry Fields](https://strawberryfields.readthedocs.io/en/latest/code/backend.states.html?highlight=poly_quad#strawberryfields.backends.states.BaseFockState.poly_quad_expectation).

* As you mention above, let's also rename `NumberStateProjector` to `FockStateProjector`, to bring it in line with the rest of the terminology (since we use 'Fock state' in all other places)

**Comment by johannesjmeyer** [comment]
> Before merging, a few things that I would like to finalize:
> 
> * Let's rename `QuadratureOperator`  to `QuadOperator` - this matches the terminology used in [OpenFermion](https://openfermion.readthedocs.io/en/latest/openfermion.html#openfermion.ops.QuadOperator) and [Strawberry Fields](https://strawberryfields.readthedocs.io/en/latest/code/backend.states.html?highlight=poly_quad#strawberryfields.backends.states.BaseFockState.poly_quad_expectation).
> * As you mention above, let's also rename `NumberStateProjector` to `FockStateProjector`, to bring it in line with the rest of the terminology (since we use 'Fock state' in all other places)

Done!

File: .github/CHANGELOG.md [hunk 2]
```diff
@@ -100,9 +103,7 @@
 
 This release contains contributions from:
 
-Shahnawaz Ahmed, riveSunder, Aroosa Ijaz, Josh Izaac, Nathan Killoran, Maria Schuld.
-
-
+Shahnawaz Ahmed, riveSunder, Aroosa Ijaz, Josh Izaac, Nathan Killoran, Maria Schuld, Johannes Jakob Meyer.
```
[the remark this answers is not in the document]
Comment (johannesjmeyer) [reply .github/CHANGELOG.md]: Okay, then I'll just remove it for now and re-add it after the merge (in front of Maria :))

File: doc/code/templates.rst [hunk 3]
```diff
@@ -178,7 +178,7 @@ the :func:`~.interferometer_uniform` function.
         for w in range(n_wires):
             qml.Squeezing(r[w][0], r[w][1], wires=w)
         Interferometer(theta=theta, phi=phi, varphi=varphi, wires=range(n_wires))
-        return [qml.expval.MeanPhoton(wires=w) for w in range(n_wires)]
+        return [qml.expval.NumberOperator(wires=w) for w in range(n_wires)]
```
[the remark this answers is not in the document]
Comment (johannesjmeyer) [reply doc/code/templates.rst]: Okay, then I'll have a look and resolve them (hopefully) :)

**Comment by johannesjmeyer** [comment]
> Would you mind making an issue over in PennyLane-SF, linking to this PR, and reminding us to make the equivalent changes?

Will do!

**Comment by johannesjmeyer** [comment]
I would like to postpone the Merge of this PR until the other PRs are merged (especially device.sample).

**Pull request merged**